src/Microsoft.AspNetCore.Razor.Tools/RequestDispatcher.cs

```diff
@@ -34,427 +34,434 @@ public static RequestDispatcher Create(ConnectionHost connectionHost, CompilerHo
         public static readonly TimeSpan GCTimeout = TimeSpan.FromSeconds(30);
 
         public abstract void Run();
+    }
 
-        private enum State
+    // Internal for testing
+    internal class DefaultRequestDispatcher : RequestDispatcher
```

Comment:
If this is going to be internal and non-nested pull it out into it's own file

Reply:
BTW because you changed the nesting, everything in this class shows up in the diff, so it's pretty hard to review.

Reply:
Sorry about that. This should help https://github.com/aspnet/Razor/pull/1969/files?w=1